doc/concepts/moveit_task_constructor/moveit_task_constructor.rst

```diff
@@ -0,0 +1,226 @@
+
+* ``CartesianPath`` - Moves the end effector in a straight line in Cartesian space.
+
+Code Example on how to initialize the solver
+
+.. code-block:: c++
+
+  const auto mtc_pipeline_planner = std::make_shared<moveit::task_constructor::solvers::PipelinePlanner>(
+      node, "ompl", "RRTConnectkConfigDefault");
+  const auto mtc_joint_interpolation_planner =
+      std::make_shared<moveit::task_constructor::solvers::JointInterpolationPlanner>();
+  const auto mtc_cartesian_planner = std::make_shared<moveit::task_constructor::solvers::CartesianPath>();
+
+These solvers will be passed into stages like ``MoveTo``, ``MoveRelative``, and ``Connect``.
+
+Setting Properties
+------------------
+
+| Each MTC stage has configurable properties. Example - planning group, timeout, goal state, etc.
+| Properties of different types can be set using the function below.
+
+.. code-block:: c++
+
+  void setProperty(const std::string& name, const boost::any& value);
+
+| Children stages can easily inherit properties from their parents, thus reducing the configuration overhead.
+
+Cost calculator for Stages
+---------------------------
+
+CostTerm is the basic interface to compute costs for solutions for MTC stages.
+
+CostTerm implementations available in MTC
+
+* ``Constant`` - Adds a constant cost to each solution
+
+* ``PathLength`` - Cost depends on trajectory length with optional weight for different joints
+
+* ``TrajectoryDuration`` - Cost depends on execution duration of the whole trajectory
+
+* ``TrajectoryCostTerm`` - cost terms that only work on SubTrajectory solutions
+
+* ``LambdaCostTerm`` - Pass in a lambda expression to calculate cost
+
+* ``DistanceToReference`` - Cost depends on weighted joint space distance to a reference point
+
+* ``LinkMotion`` - Cost depends on length of Cartesian trajectory of a link
+
+* ``Clearance`` - Cost is inverse of distance to collision
+
+Example code on how to set CostTerm using ``LambdaCostTerm``
+
+.. code-block:: c++
+
+  stage->setCostTerm(moveit::task_constructor::LambdaCostTerm(
+        [](const moveit::task_constructor::SubTrajectory& traj) { return 100 * traj.cost(); }));
+
+All stages provided by MTC have default cost terms. Stages which produce trajectories as solutions usually use path length to calculate cost.
+
+Planning and Executing a MTC Task
+---------------------------------
+
+Planning an MTC task will return a ``MoveItErrorCode``.
+Refer :moveit_msgs_codedir:`here<msg/MoveItErrorCodes.msg>` to identity the different error types.
+If planning succeeds, you can expect the plan function to return ``moveit_msgs::msg::MoveItErrorCodes::SUCCESS``.
+
+.. code-block:: c++
+
+  auto error_code = task.plan()
+
+After planning, extract the first successful solution and pass it to the execute function. This will create an ``execute_task_solution`` action client and the action server resides in ``execute_task_solution_capability`` plugin provided by MTC.
```

Comment:
```suggestion
After planning, extract the first successful solution and pass it to the execute function. This will create an ``execute_task_solution`` action client.
The action server resides in the ``execute_task_solution_capability`` plugin provided by MTC.
```